packages/packages/externalize-wordpress-assets-webpack-plugin/src/index.ts

```diff
@@ -0,0 +1,116 @@
+import { ExternalsPlugin, Compiler } from 'webpack';
+
+type ExternalsMap = {
+	exact: Record< string, string >;
+	startsWith: Record< string, string >;
+}
+
+type NormalizedOptions = {
+	referenceKey: string,
+	externalsMap: ExternalsMap
+}
+
+type Options = {
+	referenceKey: string,
+	externalsMap?: Partial<ExternalsMap>
+}
+
+const baseExternalsMap: ExternalsMap = {
+	exact: {
+		react: 'React',
+		'react-dom': 'ReactDOM',
+	},
+	startsWith: {
+		'@elementor/': '__UNSTABLE__elementorPackages',
+		'@wordpress/': 'wp',
+	},
+};
+
+export class ExternalizeWordPressAssetsWebpackPlugin {
+	externalPlugin: ExternalsPlugin;
+
+	options: NormalizedOptions;
+
+	constructor( options: Options ) {
+		this.options = this.normalizeOptions( options );
+
+		this.externalPlugin = new ExternalsPlugin(
+			'window',
+			( { request }, callback ) => {
+				const externalRequest = this.requestToExternal( request );
+
+				if ( externalRequest ) {
+					return callback( undefined, externalRequest );
+				}
+
+				return callback();
+			}
+		);
+	}
+
+	apply( compiler: Compiler ) {
+		this.externalPlugin.apply( compiler );
+
+		compiler.hooks.environment.tap( this.constructor.name, () => {
+			this.exportAsWindowItem( compiler );
+		} );
+	}
+
+	requestToExternal( request: string | undefined ) {
+		if ( ! request ) {
+			return;
+		}
+
+		const { exact, startsWith } = this.options.externalsMap;
+
+		if ( Object.keys( exact ).includes( request ) ) {
+			return exact[ request ];
+		}
+
+		for ( const [ prefix, namespace ] of Object.entries( startsWith ) ) {
```

Comment:
Shouldn't it be one of those?

```suggestion
		for ( const [ prefix, referenceKey ] of Object.entries( startsWith ) ) {
```

```suggestion
		for ( const [ prefix, globalKey ] of Object.entries( startsWith ) ) {
```

Reply:
Done